app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryAddWordContents.java

```diff
@@ -0,0 +1,363 @@
+    }
+
+    UserDictionaryAddWordContents(final View view, final UserDictionaryAddWordContents oldInstanceToBeEdited) {
+        mModeTitle = view.findViewById(R.id.user_dictionary_mode_title);
+        mWordEditText = view.findViewById(R.id.user_dictionary_add_word_text);
+        mShortcutEditText = view.findViewById(R.id.user_dictionary_add_shortcut);
+        mWeightEditText = view.findViewById(R.id.user_dictionary_add_weight);
+
+        mOldWord = oldInstanceToBeEdited.mSavedWord;
+        mOldShortcut = oldInstanceToBeEdited.mSavedShortcut;
+        mOldWeight = oldInstanceToBeEdited.mSavedWeight;
+        updateLocale(mLocale);
+    }
+
+    // locale may be null, this means default locale
+    // It may also be the empty string, which means "all locales"
+    void updateLocale(final String locale) {
+        mLocale = null == locale ? Locale.getDefault().toString() : locale;
+        // The keyboard uses the language layout of the user dictionary
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            mWordEditText.setImeHintLocales(new LocaleList(LocaleUtils.constructLocaleFromString(mLocale)));
+        }
+
+    }
+
+    void delete(final Context context) {
+        final ContentResolver resolver = context.getContentResolver();
+        final String localeInToast = new LocaleRenderer(context, mLocale).toString();
+        final String messageDeleted = context.getString(R.string.user_dict_word_deleted) + " " + localeInToast;
+        // Mode edit: remove the old entry.
+        if (MODE_EDIT == mMode && !TextUtils.isEmpty(mOldWord)) {
+            UserDictionarySettings.deleteWordInEditMode(mOldWord, mOldShortcut, mOldWeight, resolver);
+            // Toast appears to indicate that the word has been deleted
+            Toast.makeText(context, messageDeleted, Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    public final int apply(@NonNull final Context context) {
+        final ContentResolver resolver = context.getContentResolver();
+        final String newWord = mWordEditText.getText().toString();
+        final String newShortcut;
+        final String newWeight;
+
+        if (null == mShortcutEditText) {
+            newShortcut = null;
+        } else {
+            final String tmpShortcut = mShortcutEditText.getText().toString();
+            if (TextUtils.isEmpty(tmpShortcut)) {
+                newShortcut = null;
+            } else {
+                newShortcut = tmpShortcut;
+            }
+        }
+
+        if (mWeightEditText == null) {
+            newWeight = String.valueOf(WEIGHT_FOR_USER_DICTIONARY_ADDS);
+        } else {
+            final String tmpWeight = mWeightEditText.getText().toString();
+            if (TextUtils.isEmpty(tmpWeight)) {
+                newWeight = String.valueOf(WEIGHT_FOR_USER_DICTIONARY_ADDS);
+            } else {
+                newWeight = tmpWeight;
+            }
+        }
+
+        if (TextUtils.isEmpty(newWord)) {
```

Comment:
This could be moved further up, there is no reason to do anything else (like checking shortcut text) when the word is empty.

Reply:
Change made with commit 0582df5